play-services-core/src/main/java/org/microg/gms/gcm/McsInputStream.java

```diff
@@ -57,7 +56,7 @@
     private int streamId = 0;
     private long lastMsgTime = 0;
 
-    private boolean closed = false;
+    private volatile boolean closed = false;
```

Comment:
This gets accessed from more than one thread, so at the very least it needs to be `volatile`.